engine/storage/volume/src/main/java/org/apache/cloudstack/storage/datastore/PrimaryDataStoreImpl.java

```diff
@@ -91,6 +91,9 @@ public class PrimaryDataStoreImpl implements PrimaryDataStore {
     private VolumeDao volumeDao;
     private Map<String, String> _details;
 
+    private String uuid;
```

Comment:
why the changes in this file? they seem to only add maintenance risk.

Reply:
@DaanHoogland the changes here were made in order to have the proper information to identify the objects in the log messages

Reply:
The extra fields are not exposed and the `psdv` is available to retrieve the `uuid` and `name`, so why would you store the fields extra? the old implementation of `getUuid()` and `getName()` seems to suffice.

Reply:
@DaanHoogland, sorry, my explanation was not clear in the first comment.

Those fields were added this way because we're using `ReflectionToStringBuilderUtils` to build the string to print in the log. As `ReflectionToStringBuilderUtils` uses the object attributes to build the string, this object is declared as an interface `StoragePool` (in runtime this object is an instance of `PrimaryDataStoreImpl`) and we cannot retrieve the `StoragePoolVO` contained in it, we declared these attributes (that already are there) to be reflected later.

Reply:
ok, acceptable ;)